src/EditorFeatures/Test2/Diagnostics/DiagnosticServiceTests.vb

```diff
@@ -1940,8 +1940,15 @@ class MyClass
                 ' Get cached project diagnostics.
                 Dim diagnostics = diagnosticService.GetCachedDiagnosticsAsync(workspace, project.Id).WaitAndGetResult(CancellationToken.None)
 
-                Assert.Equal(1, diagnostics.Count())
-                Assert.Equal(HiddenDiagnosticsCompilationAnalyzer.Descriptor.Id, diagnostics.Single().Id)
+                ' different behavior between v1 and v2. in v2. in v2, solution crawler never creates non-local hidden diagnostics.
```

Comment:
I think we shouldn't support v1 behavior until we sort out what to do with it. since v1 is more costly than this behavior. I think until we have confirm idea on what we actually want from it, we should always take cheaper one.